code/obj/item/organs/stomach.dm

```diff
@@ -7,42 +7,86 @@
 	organ_holder_required_op_stage = 4
 	icon_state = "stomach"
 	fail_damage = 100
+	///How much food can we fit, based on `fill_amt` var on food items
+	var/capacity = 7
+	///How much food and other stuff we have (also based on `fill_amt`)
+	var/food_amount = 0
+	///Stomach contents are actually stored in the mob so that things like matsci effects work
+	VAR_PRIVATE/atom/movable/stomach_contents = list()
 
-	//Do something with this when you figure out what the guy who made digestion and handle stomach was doing with stomach_contents and stomach_process - kyle
-	// on_transplant()
-	// 	..()
-	// 	if (iscarbon(src.donor))
-	// 		src.donor.stomach_contents = src.contents
-	// 		src.contents = null //Probably don't need to do this, will undo if I ever remove the var off of mob and into stomach completely -kyle
-	// on_removal()
-	// 	..()
-	// 	//Add stomach contents on mob to this object for transplants
-	// 	if (iscarbon(src.donor))
-	// 		src.contents = src.donor.stomach_contents
-	// 		src.donor.stomach_contents = src.donor.stomach_contents.Cut()
-
-//
 	on_transplant()
 		..()
-		if (iscarbon(src.donor))
-			src.donor.stomach_process = src.contents
-			src.contents = list() //Probably don't need to do this, will undo if I ever remove the var off of mob and into stomach completely -kyle
-		// if (src.donor)
-			// for (var/datum/ailment_data/disease in src.donor.ailments)
-			// 	if (disease.cure == "Stomach Transplant")
-			// 		src.donor.cure_disease(disease)
-			// return
+		for (var/atom/movable/AM in src.stomach_contents)
+			AM.set_loc(src.donor)
+		if (src.is_full())
+			src.donor.setStatus("full")
+
 	on_removal()
-		//Add stomach contents on mob to this object for transplants
-		if (iscarbon(src.donor))
-			src.contents = src.donor.stomach_process
-			src.donor.stomach_process = list()
+		for (var/atom/movable/AM in src.stomach_contents)
+			AM.set_loc(src) // take them with us
+		src.donor.delStatus("full")
 		..()
 
+	attackby(obj/item/W, mob/user)
+		if (iscuttingtool(W))
+			user.visible_message("<span class='alert'>[user] starts cutting [src] open!")
+			SETUP_GENERIC_ACTIONBAR(user, src, 4 SECONDS, PROC_REF(cut_open), list(), W.icon, W.icon_state, "[user] cuts [src] open, spilling its contents everywhere!", INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_STUNNED | INTERRUPT_ACTION)
+			return
+		. = ..()
+
+	proc/cut_open()
+		for (var/atom/movable/AM in src.stomach_contents)
+			AM.set_loc(get_turf(src))
+		src.stomach_contents = null
+		src.splat(get_turf(src))
+		qdel(src)
+
+	///How much does this thing fill a stomach
+	proc/food_value(atom/movable/AM)
+		if (istype(AM, /obj/item/reagent_containers/food))
+			var/obj/item/reagent_containers/food/food = AM
+			return food.fill_amt
+		else
+			return 1 //other stuff can clog your stomach
+
+	proc/consume(atom/movable/AM)
+		AM.set_loc(src.donor)
+		src.stomach_contents |= AM
+		src.food_amount += src.food_value(AM)
```

Comment:
probably not necessary, but you could maybe do a length check before & after to see if `food_amount` needs incrementing. This would only be useful if it was in some way possible to eat things that are already in your stomach, but gooncode does like to surprise.